### PR TITLE
Added confirmation before deleting comment

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #3547

## Description of Changes
- Adds confirmation popup when deleting comment

## Test Plan
- CA (click around) tested on local

## Other Considerations
It looks like this:
Screenshot 2023-04-26 at 12 40 45 PM

Let me know if the design needs changing.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/comments/comment.tsx [hunk 1]
```diff
@@ -5,8 +5,9 @@ import type { Account, Comment as CommentType } from 'models';
 import moment from 'moment';
 
 import app from 'state';
-import { ContentType } from 'types';
+import { ContentType, ThreadActionType } from 'types';
 import { ChainType } from '../../../../../../common-common/src/types';
+import deleteComment from '../../../../../server/routes/deleteComment';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/comments/comment.tsx:10]: We should not be importing routes from the server onto the client.

Also can we make the common-common import just `'common-common/src/types'`?

File: packages/commonwealth/client/scripts/views/components/comments/comment.tsx [hunk 2]
```diff
@@ -100,9 +101,30 @@ export const Comment = (props: CommentProps) => {
...
+    openConfirmation({
+      title: 'Delete Comment',
+      description: <>Delete this comment?</>,
+      buttons: [
+        {
+          label: 'Delete',
+          buttonType: 'mini-red',
+          onClick: async () => {
+            try {
+              app.comments.delete(comment).then(() => {
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/comments/comment.tsx:114]: Any reason why we're using .then() inside an async function?

File: packages/commonwealth/client/scripts/views/components/comments/comment.tsx [hunk 3]
```diff
@@ -100,9 +101,30 @@ export const Comment = (props: CommentProps) => {
...
+                updatedCommentsCallback();
+              });
+            } catch (err) {
+              console.log(err);
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/comments/comment.tsx:118]: We don't surface errors to the user? We should do a toast.

**Review by jnaviask — CHANGES_REQUESTED** [review]
Cleanup.

File: packages/commonwealth/client/scripts/views/components/comments/comment.tsx (hunk 1 above)
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/components/comments/comment.tsx:10]: My autoimport pulled it in by accident when I was typing

Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/components/comments/comment.tsx]: Can we fix the other import while we're in this file?

**Review by jnaviask — APPROVED** [review]

**Pull request merged**